src/main/java/org/jetbrains/bsp/bazel/server/bsp/resolvers/actiongraph/ActionGraphV2Parser.java

```diff
@@ -0,0 +1,96 @@
+package org.jetbrains.bsp.bazel.server.bsp.resolvers.actiongraph;
+
+import com.google.common.collect.Lists;
+import com.google.devtools.build.lib.analysis.AnalysisProtosV2;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ActionGraphV2Parser extends ActionGraphParser {
+  private final AnalysisProtosV2.ActionGraphContainer actionGraph;
+  private final Map<Integer, AnalysisProtosV2.PathFragment> pathFragmentMap;
+
+  public ActionGraphV2Parser(AnalysisProtosV2.ActionGraphContainer actionGraph) {
+    this.actionGraph = actionGraph;
+    pathFragmentMap =
+        actionGraph.getPathFragmentsList().stream()
+            .collect(Collectors.toMap(AnalysisProtosV2.PathFragment::getId, fragment -> fragment));
+  }
+
+  @Override
+  protected Stream<String> buildInputs(String target, List<String> suffixes) {
+    return getActions(target).stream()
+        .flatMap(action -> action.getInputDepSetIdsList().stream())
+        .flatMap(depset -> expandDepsetToArtifacts(depset).stream())
+        .map(AnalysisProtosV2.Artifact::getPathFragmentId)
+        .map(pathFragmentId -> EXEC_ROOT + constructPath(pathFragmentId));
+  }
+
+  private String constructPath(Integer pathFragmentId) {
```

Comment:
I would love to see a more informative name for this method. What kind of path it constructs?

Reply:
I'm not sure if I can a better one as of right now... This function is quite generic in the sense that it builds a path from a path fragment, which is a way of an artifact expressing a portion of its path to the file it represents. Can you think of a way of naming this better?